### PR TITLE
fix: Solve the problem of failure to download multi-branch-pipeline artifacts

### PR DESCRIPTION
**Pull request by littlejiancc**
### What type of PR is this?
/kind bug

### What this PR does / why we need it:
Multi-branch pipeline artifacts download failed
jenkins-client pr: https://github.com/jenkins-zh/jenkins-client/pull/64

### Which issue(s) this PR fixes:

Fixes #https://github.com/kubesphere/kubesphere/issues/5821

### Special notes for reviewers:

Please check the following list before waiting reviewers:

- [ ] Already committed the CRD files to [the Helm Chart](https://github.com/kubesphere-sigs/ks-devops-helm-chart/) if you created some new CRDs
- [ ] Already [added the permission](https://github.com/kubesphere/ks-installer/blob/9e063b085a0e43fdb3d0d9e3e7f4149146f14b9c/roles/ks-core/prepare/files/ks-init/role-templates.yaml) for the new API
- [ ] Already added the RBAC markers for the new controllers

### Does this PR introduce a user-facing change??
None

```release-note
fixed the bug with downloading multi-branch-pipeline artifacts

```

### CODE REVIEW

**Comment by littlejiancc** [comment]
@chilianyi done

File: pkg/kapis/devops/v1alpha3/pipelinerun/handler.go [hunk 1]
```diff
@@ -233,9 +234,11 @@ func (h *apiHandler) downloadArtifact(request *restful.Request, response *restfu
 		return
 	}
 	pipelineName := pr.Labels[v1alpha3.PipelineNameLabelKey]
+	isMultiBranch := pr.Spec.IsMultiBranchPipeline()
+	branchName := pr.GetRefName()
```
Comment (littlejiancc) [inline pkg/kapis/devops/v1alpha3/pipelinerun/handler.go:238]: Not sure if this is the right way to get branchName.

Comment (chilianyi) [reply pkg/kapis/devops/v1alpha3/pipelinerun/handler.go:238]: Yes, it is right.

**Comment by chilianyi** [comment]
> @chilianyi done

Thanks for your pr. 

Please merge the commit records through git rebase, now there are five commit, it will be better to merge them into one commit.

**Comment by littlejiancc** [comment]
> 
Even if there are 5 commits, but the pr is combined into one.

**Comment by littlejiancc** [comment]
> > 
> 
> Even if there are 5 commits, but the pr is combined into one.

**Comment by littlejiancc** [comment]
> > @chilianyi done
> 
> Thanks for your pr.
> 
> Please merge the commit records through git rebase, now there are five commit, it will be better to merge them into one commit.

Even if there are 5 commits, but the pr is combined into one.

File: pkg/kapis/devops/v1alpha3/pipelinerun/handler.go (hunk 1 above)
Comment (littlejiancc) [reply pkg/kapis/devops/v1alpha3/pipelinerun/handler.go:238]: thx

**Comment by chilianyi** [comment]
> Even if there are 5 commits, but the pr is combined into one.

Yes, but this helps to keep the commit history clean and concise, making it clearer when viewing the changelog directly.
Thanks.

**Comment by littlejiancc** [comment]
> Yes, but this helps to keep the commit history clean and concise, making it clearer when viewing the changelog directly. Thanks.

Ok, I understand, and done it.

**Comment by littlejiancc** [comment]
@yudong2015 Is it possible to cherry-pick to release-3.3?
Because the artifact download function is already available in release-3.3

**Pull request merged**